### PR TITLE
Expand included collection items

### PR DESCRIPTION
**Pull request by pape-io**
In the following case, LHS was not expanding included resources:

```
Customer.includes(:places).find(1)
GET 'customers/1'
```
```
{
  places: { href: 'customers/1/places' }
}
```
`includes` fetches places
```
GET 'customers/1/places'
{
  items: [
    { href: 'datastore/places/1' },
    { href: 'datastore/places/2' },
    { href: 'datastore/places/3' }
  ]
}
```
but this just includes a list of links, so LHS was not further including the requested resources.

Now it does also fetch all those links in order to fulfil the requested: `Customer.includes(:places).find(1)`

Backwards compatible: Minor

### CODE REVIEW

**Comment by cedricwider** [comment]
👍

File: lib/lhs/concerns/record/request.rb [hunk 1]
```diff
@@ -76,7 +76,14 @@ def extend_base_item_with_collection!(data, addition, key)
         if target._raw.is_a? Array
           data[key] = addition.map(&:_raw)
         else # hash with items
-          target._raw[items_key] = addition.map(&:_raw)
+          target._raw[items_key] ||= []
+          if target._raw[items_key].empty?
```
Comment (Averethel) [inline lib/lhs/concerns/record/request.rb]: We have some nesting here. Do you think it makes sense to extract this to auxiliary method?

File: lib/lhs/concerns/record/request.rb [hunk 2]
```diff
@@ -92,16 +99,42 @@ def handle_includes(includes, data)
...
-            url_option_for(data, included)
-          end
+        options = options_for_data(data, included)
         addition = load_include(options, data, sub_includes)
         extend_raw_data!(data, addition, included)
+        expand_addition!(data, included) if no_expanded_data?(addition)
+      end
+
+      def options_for_data(data, included = nil)
+        if data.collection?
```
Comment (Averethel) [inline lib/lhs/concerns/record/request.rb]: I'd rather go for guard clauses:
```ruby
return options_for_multiple(data, included) if data.collection?
return options_for_nested_items(data, included) if included && data[included].collection?
url_option_for(data, included)
```
What do you think?

**Comment by Averethel** [comment]
👍

File: lib/lhs/concerns/record/request.rb (hunk 1 above)
Comment (pape-io) [inline lib/lhs/concerns/record/request.rb]: done 👍

File: lib/lhs/concerns/record/request.rb (hunk 2 above)
Comment (pape-io) [inline lib/lhs/concerns/record/request.rb]: 👍

**Pull request merged**